pkg/distcache/provider.go

```diff
@@ -172,11 +233,28 @@ func (constructor DistCacheConstructor) ProvideDistCache(in DistCacheConstructor
 	}
 
 	dc.Olric = o
+	job := &jobs.BasicJob{
+		JobBase: jobs.JobBase{
+			JobName: distCacheMetricsJobName,
+		},
+		JobFunc: dc.scrapeMetrics,
+	}
+	dc.Metrics = newDistCacheMetrics()
 
 	in.Lifecycle.Append(fx.Hook{
 		OnStart: func(ctx context.Context) error {
+			// Register metrics with Prometheus.
+			err := dc.Metrics.registerMetrics(in.PrometheusRegistry)
+			if err != nil {
+				log.Error().Err(err).Msg("Failed to register distcache metrics with Prometheus registry")
+				return err
+			}
+
 			panichandler.Go(func() {
 				err = dc.Olric.Start()
```

Comment:
would be good to use a local variable for error as it might interfere with variable outside this go-routine.

Reply:
I had to change the error name, `startErr`; otherwise, golangci yells due to shadow declaration